src/fromager/extras_provider.py

```diff
@@ -18,37 +18,36 @@
 this, we constrain the solution to require a unique version of X.
 """
 
+from typing import TYPE_CHECKING
+
+from packaging.requirements import Requirement
 from resolvelib.providers import AbstractProvider
 
+from .candidate import Candidate
+
+# fix for runtime errors caused by inheriting classes that are generic in stubs but not runtime
+# https://mypy.readthedocs.io/en/latest/runtime_troubles.html#using-classes-that-are-generic-in-stubs-but-not-at-runtime
+if TYPE_CHECKING:
+    BaseAbstractProvider = AbstractProvider[Requirement, Candidate, str]
+else:
+    BaseAbstractProvider = AbstractProvider
 
-class ExtrasProvider(AbstractProvider):
+
+class ExtrasProvider(BaseAbstractProvider):
     """A provider that handles extras."""
 
-    def get_extras_for(self, requirement_or_candidate):
+    def get_extras_for(
+        self, requirement_or_candidate: Requirement | Candidate
+    ) -> tuple[str]:
         """Given a requirement or candidate, return its extras.
 
         The extras should be a hashable value.
         """
         raise NotImplementedError
 
-    def get_base_requirement(self, candidate):
+    def get_base_requirement(self, candidate: Candidate) -> Requirement:
         """Given a candidate, return a requirement that specifies that
         project/version.
 
         """
         raise NotImplementedError
-
-    def identify(self, requirement_or_candidate):
-        base = super().identify(requirement_or_candidate)
-        extras = self.get_extras_for(requirement_or_candidate)
-        if extras:
-            return (base, extras)
-        else:
-            return base
-
-    def get_dependencies(self, candidate):
-        deps = super().get_dependencies(candidate)
-        if candidate.extras:
-            req = self.get_base_requirement(candidate)
-            deps.append(req)
-        return deps
```

Comment:
This method calls `super().get_dependencies` but that will raise `NotImplementedError` since https://github.com/sarugaku/resolvelib/blob/main/src/resolvelib/providers.py#L132

Since we have not seen this error I think we can safely remove it

Reply:
SGTM